### PR TITLE
Nictotine Regeant Effects

### PR DESCRIPTION
**Pull request by MaxIsJoe**
CL: [New] Nicotine reagents now add a speed buff to bodies when flowing into a creature's blood stream.
CL: [New] Smoking for prolonged periods of time results in severe lung damage. Take breaks in-between drags.

### CODE REVIEW

File: UnityProject/Assets/Scripts/ScriptableObjects/RP/EmoteBehaviors/GiveSpeedStatusEffect.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿using Systems.StatusesAndEffects;
+using UnityEngine;
+
+namespace ScriptableObjects.RP.EmoteBehaviors
+{
+	public class GiveSpeedStatusEffect : IEmoteBehavior
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/ScriptableObjects/RP/EmoteBehaviors/GiveSpeedStatusEffect.cs:6]: why an emote? Where are you calling this emote?

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/ScriptableObjects/RP/EmoteBehaviors/GiveSpeedStatusEffect.cs:6]: `CoughAndGiveSpeed.asset`
The emote is getting called in `TobaccoInBody.asset` as its an reaction that triggers emotes.

Comment (corp-0) [reply UnityProject/Assets/Scripts/ScriptableObjects/RP/EmoteBehaviors/GiveSpeedStatusEffect.cs:6]: ah, fair

File: UnityProject/Assets/Scripts/Systems/StatusesAndEffects/Implementations/SpeedBuff.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using Logs;
+using Player.Movement;
+using Systems.StatusesAndEffects.Interfaces;
+using UnityEngine;
+
+namespace Systems.StatusesAndEffects.Implementations
+{
+	[CreateAssetMenu(fileName = "Speed Buff", menuName = "ScriptableObjects/StatusEffects/SpeedBuff")]
+	public class SpeedBuff : StatusEffect, IExpirableStatus, IMovementEffect, IStackableStatus
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Systems/StatusesAndEffects/Implementations/SpeedBuff.cs]: imo Would have been nice if it was checking for the nicotine reagent in the blood and then did stuff from that?

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Systems/StatusesAndEffects/Implementations/SpeedBuff.cs]: but its already doing that?

**Pull request merged**